constraints.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+package magistrala
+
+type Operation int
+
+const (
+	Create Operation = iota
+)
+
+// Constraints specifies an API for obtaining entity constraints.
+type Constraints interface {
+	// Constraints returns constraints for the entities.
+	CheckLimits(operation Operation, currentValue uint64) error
```

Comment:
Is there any other type of limit that will not be `uint64`? I suspect at least float64 is a better fit, but maybe we should consider `any` as well.